cranelift/filetests/src/test_run_zkasm.rs

```diff
@@ -0,0 +1,182 @@
+
+use cranelift_codegen::ir::function::FunctionParameters;
+use cranelift_codegen::ir::ExternalName;
+use cranelift_codegen::isa::zkasm;
+use cranelift_codegen::{settings, FinalizedMachReloc, FinalizedRelocTarget};
+
+struct TestRunZkasm;
+
+pub fn subtest(parsed: &TestCommand) -> anyhow::Result<Box<dyn SubTest>> {
+    assert_eq!(parsed.command, "run-zkasm");
+    if !parsed.options.is_empty() {
+        anyhow::bail!("No options allowed on {}", parsed);
+    }
+    Ok(Box::new(TestRunZkasm))
+}
+
+impl SubTest for TestRunZkasm {
+    fn name(&self) -> &'static str {
+        "run-zkasm"
+    }
+
+    fn is_mutating(&self) -> bool {
+        false
+    }
+
+    fn needs_isa(&self) -> bool {
+        false
+    }
+
+    /// Runs the entire subtest for a given target, invokes [Self::run] for running
+    /// individual tests.
+    fn run_target<'a>(
+        &self,
+        testfile: &TestFile,
+        _: &mut FileUpdate,
+        _: &'a str,
+        _: &'a Flags,
+        _: Option<&'a dyn TargetIsa>,
+    ) -> anyhow::Result<()> {
+        // We can build the FunctionStore once and reuse it
+        let mut func_store = FunctionStore::default();
+        for (func, _) in &testfile.functions {
+            func_store.add(func.name.to_string(), &func);
+        }
+
+        for (func, details) in &testfile.functions {
+            info!("Test: {}({}) runner-zkasm", self.name(), func.name);
+
+            run_test(&func_store, func, details).context(self.name())?;
+        }
+
+        Ok(())
+    }
+
+    fn run(
+        &self,
+        _func: Cow<ir::Function>,
+        _context: &crate::subtest::Context,
+    ) -> anyhow::Result<()> {
+        unreachable!()
+    }
+}
+
+// TODO: replace by results of https://github.com/near/wasmtime/pull/213
+struct ZkAsmExecutor {}
+
+impl ZkAsmExecutor {
+    pub fn new() -> ZkAsmExecutor {
+        ZkAsmExecutor {}
+    }
+
+    pub fn execute(self, zkasm_function: &Vec<String>, args: &Vec<DataValue>) -> DataValue {
+        for line in zkasm_function {
+            println!("{}", line);
+        }
+        for arg in args {
+            println!("{:#?}", arg);
+        }
+        todo!();
+    }
+}
+
+fn run_test(func_store: &FunctionStore, func: &Function, details: &Details) -> anyhow::Result<()> {
+    let zkasm_function = generate_zkasm(func_store, func);
+    for comment in details.comments.iter() {
+        if let Some(command) = parse_run_command(comment.text, &func.signature)? {
+            trace!("Parsed run command: {}", command);
+            let _ = command.run(|_func_name, run_args| {
+                let mut args = Vec::with_capacity(run_args.len());
+                args.extend_from_slice(run_args);
+                let executor = ZkAsmExecutor::new();
+
+                // just wrap to make CI green until executor is not ready
+                // TODO: replace by actual execution and results checking
+                let _ = panic::catch_unwind(AssertUnwindSafe(|| {
```

Comment:
From once side yes, it will simplify, from the other side, this annoying ``catch_unwind`` and ``todo!`` explicitly marks that this parts are not ready and should be rewritten